[10.0][REM][ADD] Deprecate medical.physician in favor of medical.practitioner

obulkin opened the pull request:
This PR depends on:

- [x] https://github.com/LasLabs/vertical-medical/pull/253

obulkin committed:
[REM][ADD] medical_physician: Deprecate physician
* Remove data and demo records, default physician avatars, views, wizards, and
all models other than medical.physician
* Remove schedule_template_ids field from medical.physician and add
equiv_practitioner_id field for storing equivalent medical practitioner
* Overload create in medical.physician to prevent creation of new records
* Overload write in medical.physician to log error when a record is modified
* Add migration logic that creates a new medical.practitioner record for every
physician and performs various clean up
* Update access rules, manifest, README, and unit tests accordingly

obulkin committed:
[REM] medical_physician_us: Deprecate physician
* Remove data records, model, tests, and views
* Update manifest and README accordingly

lasley commented on medical_physician/migrations/10.0.2.0.0/post-migrate.py:
```diff
@@ -0,0 +1,76 @@
+def migrate(cr, version):
+    """Create practitioner record for each physician and clean up"""
+    env = api.Environment(cr, SUPERUSER_ID, {})
+    physicians = env['medical.physician'].search([])
```
Will need a domain to include inactive.

```
'|',
('active', '=', True),
('active', '=', False),
```

lasley commented on medical_physician/migrations/10.0.2.0.0/post-migrate.py:
```diff
@@ -0,0 +1,76 @@
+
+    # Use new specialty since old record will be removed
+    old_gp_specialty = env.ref('medical_physician.medical_specialty_gp')
+    gp_specialty = env.ref('medical_practitioner.medical_specialty_gp')
```
For uniformity I think this should be `new_gp_specialty`.

But for double awesomeness, the names should be reversed a bit. When naming grouped/similar objects, think high level then drill to granular. In this  example:

```
specialty_gp_new
specialty_gp_old
```

obulkin committed:
[REM][ADD] medical_patient_disease: Deprecate physician
* Remove physician_id field from demo data and views
* Add practitioner_id field to medical.patient.disease model and views
* Add migration logic that connects every disease with a related physician to
that physician's equivalent practitioner record
* Update unit tests, manifest, and README accordingly

obulkin committed:
[REM][ADD] medical: Deprecate physician
* Replace physician reference in README with reference to practitioner model

obulkin committed:
[REM][ADD] medical_appointment: Deprecate physician
* Use pre_init_hook to prevent new module installs since it is designed around
the deprecated physician model and has been replaced by more abstract solutions
in crm/calendar_resource and product-attribute/product_service_duration
* Update manifest and README accordingly
* Update Travis config to exclude module from testing

obulkin committed:
[REM][ADD] medical_pharmacy: Deprecate physician
* Restructure menus to avoid dependence on removed medical_physician menu item

obulkin committed:
[REM][ADD] medical_medication: Deprecate physician
* Remove physician_id field from demo data and views
* Add practitioner_id field to medical.patient.medication model and views
* Add migration logic that connects every medication with a related physician
to that physician's equivalent practitioner record
* Update demo data, unit tests, manifest, and README accordingly

obulkin committed:
[REM] medical_patient_disease_allergy: Deprecate physician
* Remove reference to medical.physician demo record that is no longer available
* Update unit test to use new demo record ID

obulkin committed:
[REM][ADD] medical_prescription: Deprecate physician
* Remove physician_id field from demo data and views
* Add practitioner_id field to views and medical.prescription.order model
* Add migration logic that connects every prescription order to its physician's
equivalent practitioner record
* Update demo data, unit tests, manifest, and README accordingly
* Bring module further in line with OCA style guidelines

obulkin committed:
[REM][ADD] medical_prescription_disease: Deprecate physician
* Replace physician record used in unit test with a test practitioner

obulkin committed:
[REM] medical_prescription_merge: Deprecate physician
* Replace reference to unavailable medical.physician demo record with
medical.practitioner demo record
* Update wizard logic for handling prescriptions with different physicians to
look at practitioners instead
* Update unit tests accordingly

obulkin committed:
[REM][ADD] medical_prescription_disease_state: Deprecate physician
* Replace physician record used in unit tests with a test practitioner

obulkin committed:
[REM][ADD] medical_prescription_state_verify: Deprecate physician
* Replace medical.physician demo record with a demo practitioner and update
other demo data accordingly

obulkin committed:
[REM][ADD] sale_medical_prescription: Deprecate physician
* Add verification fields present in physician model to practitioner model
* Remove physician view
* Add migration logic that updates the verification fields on every physician's
equivalent practitioner using data from that physician
* Update demo data, unit tests, manifest, and README accordingly
* Bring module further in line with OCA style guidelines

obulkin committed:
[REM][ADD] sale_crm_medical_prescription: Deprecate physician
* Replace medical.physician demo record with a demo practitioner and update
other demo data accordingly
* Replace physician_id field in view with practitioner_id
* Update manifest and README accordingly

obulkin committed:
[REM][ADD][FIX] sale_stock_medical_prescription: Deprecate physician
* Replace medical.physician demo record with a demo practitioner and update
other demo data accordingly
* Increase qty field of prescription order line demo record to resolve issue
triggered during reinstallation of demo data and update unit tests accordingly
* Update manifest and README accordingly

obulkin committed:
[REM][ADD] medical_app_pharmacy: Deprecate physician
* Replace physician menu item with a practitioner one

obulkin committed:
[REM][ADD] sale_stock_medical_prescription_us: Deprecate physician
* Replace medical.physician demo record with a demo practitioner and update
other demo data accordingly
* Update manifest and README accordingly

obulkin committed:
[REM][ADD] website_portal_medical_prescription_order_line: Deprecate physician
* Replace medical.physician demo record with a demo practitioner and update
other demo data accordingly
* Update manifest accordingly

obulkin committed:
[REM][ADD] medical_prescription_state: Deprecate physician
* Replace view references to physician_id fields with practitioner_id fields

obulkin committed:
[REM][ADD] website_sale_medical_prescription: Deprecate physician
* Replace all references to the physician model and physician_id fields with
equivalent practitioner references
